src/client/repl/replCommands.ts

```diff
@@ -86,19 +115,85 @@ export async function registerReplCommands(
-                    document: ourResource,
+                    // document: ourResource,
+                    document: notebookDocument!.uri,
                 });
             }
         }),
     );
 }
+
+// Command for 'Enter': Conditionally call interactive.execute OR insert \n in text input box.
+export async function registerReplExecuteOnEnter(
+    disposables: Disposable[],
+    interpreterService: IInterpreterService,
+): Promise<void> {
+    disposables.push(
+        commands.registerCommand(Commands.Exec_In_REPL_Enter, async (uri: Uri) => {
+            const interpreter = await interpreterService.getActiveInterpreter(uri);
+            if (!interpreter) {
+                commands.executeCommand(Commands.TriggerEnvironmentSelection, uri).then(noop, noop);
+                return;
+            }
+
+            // Create Separate Python server to check valid command
+            const pythonServer = createPythonServer([interpreter!.path! as string]);
+
+            const activeEditor = window.activeTextEditor;
+            let userTextInput;
+            let completeCode = false;
+
+            if (activeEditor) {
+                const { document } = activeEditor;
+                userTextInput = document.getText();
+            }
+
+            // Check if userTextInput is a complete Python command
+            if (userTextInput) {
+                const stringBoolean = await pythonServer.checkValidCommand(userTextInput);
+                if (stringBoolean === 'True') {
+                    completeCode = true;
+                }
+            }
+            const editor = window.activeTextEditor;
+            // Execute right away when complete code and Not multi-line
+            if (completeCode && !isMultiLineText(editor)) {
+                await commands.executeCommand('interactive.execute');
+            } else {
+                // Insert new line on behalf of user. "Regular" monaco editor behavior
+                if (editor) {
+                    const position = editor.selection.active;
+                    const newPosition = position.with(position.line, editor.document.lineAt(position.line).text.length);
+                    editor.selection = new Selection(newPosition, newPosition);
+
+                    editor.edit((editBuilder) => {
+                        editBuilder.insert(newPosition, '\n');
+                    });
+                }
+
+                // Handle case when user enters on blank line, just trigger interactive.execute
+                if (editor && editor.document.lineAt(editor.selection.active.line).text === '') {
+                    await commands.executeCommand('interactive.execute');
+                }
+            }
+        }),
+    );
+}
+
+function isMultiLineText(textEditor: TextEditor | undefined): boolean {
+    if (textEditor) {
+        const { document } = textEditor;
+        return document.lineCount > 1;
+    }
+    return false;
```

Comment:
```suggestion
    return (textEditor?.document?.lineCount ?? 0) > 1;
```
This looks like this can be simplified.

Reply:
this is beautiful